cdi/src/main/java/org/hibernate/validator/cdi/spi/InjectingConstraintValidatorFactory.java

```diff
@@ -39,6 +45,14 @@ public InjectingConstraintValidatorFactory(BeanManager beanManager) {
 
 	@Override
 	public <T extends ConstraintValidator<?, ?>> T getInstance(Class<T> key) {
+		// we can do a simple check like :
+		//		if ( key.getPackageName().startsWith( "org.hibernate.validator.internal.constraintvalidators.bv" )
+		//				|| key.getPackageName().startsWith( "org.hibernate.validator.internal.constraintvalidators.hv" ) ) {
+		//			return super.getInstance( key );
+		//		}
+		if ( builtInConstraintValidators.contains( key ) ) {
+			return super.getInstance( key );
+		}
```

Comment:
There wasn't exactly a way to get the built-in validators as is soooo that's what I've done here.

- add a util in main module (engine) to provide a list of built-in constraint validator classes based on the ConstraintHelper that handles all constraint-validator related things there.
- passed the set of these classes to the factory and add a check to bypass CDI if a built-in constraint validator is requested. 

I've left that commented-out code as that's what I've started with to see if the approach of bypassing CDI will work and if there won't be any other unexpected errors. But then also, I wonder if we keep the check for those packages, do we even need the set? I mean, what are the odds of someone using that package? 😃

Reply:
> add a util in main module (engine) to provide a list of built-in constraint validator classes based on the ConstraintHelper that handles all constraint-validator related things there.

Makes sense, but:

1. Are you positive classes of built-in constraint validators were always being loaded before?
2. Maybe you can add some test OR a JQAssistant rule to be sure the util is up to date?


> what are the odds of someone using that package? 😃

In correct code? 0 chance.

In real code? Well... :)

At the very least, let's say it's possible in demos. We may not want to care, though.

Reply:
> > add a util in main module (engine) to provide a list of built-in constraint validator classes based on the ConstraintHelper that handles all constraint-validator related things there.
> 
> Makes sense, but:
> 
>     1. Are you positive classes of built-in constraint validators were always being loaded before?
> 
>     2. Maybe you can add some test OR a JQAssistant rule to be sure the util is up to date?

Ah, you're computing the list at runtime... Then I guess item 2 is irrelevant.

Reply:
yeah.. the downside I can think of, for this approach is that we will be creating the ConstraintHelper 2 times, once for CDI and once for the ValidatorFactory itself ...